package.json

```diff
@@ -31,28 +31,28 @@
     "eslint-plugin-flowtype": "^2.40.1",
     "extract-text-webpack-plugin": "^3.0.2",
     "file-loader": "^1.1.6",
-    "flow-bin": "^0.62.0",
```

Comment:
Flow seems to break with latest version. CC @marcianx

Reply:
With v0.68.0:

> Previously, Flow would allow you to write if (foo.unknownProp) { ... }.
Now Flow disallows testing unknown properties in conditionals. If foo is a
union type like { x: string } | { y: number }, x and y are known
properties and z would be an unknown property

```js
Property `mode` is missing in `MacroContextInterface` [1].

   src/macros.js:97:9
   97|     if (context.mode === 'text') {
               ^^^^^^^^^^^^

References:
   src/macros.js:47:6
   47|     (MacroContextInterface => (string | MacroExpansion));
            ^^^^^^^^^^^^^^^^^^^^^ [1]
```

Reply:
With v0.72.0:
```js
 - string literal `XXX` [1] is incompatible with `NODETYPE` [1] in type argument `NODETYPE` [2] of property
   `handler`.

   src/defineEnvironment.js:113:35
   113|         _environments[names[i]] = data;
                                          ^^^^

References:
   src/defineEnvironment.js:84:25
    84|     handler: EnvHandler<NODETYPE>,
                                ^^^^^^^^ [1]
   src/defineEnvironment.js:30:17
    30| type EnvHandler<NODETYPE: NodeType> = (
                        ^^^^^^^^ [2]

 - string literal XXX [1] is incompatible with `NODETYPE` [1] in type argument `TYPE` [2] of the return value of
   property `handler`.

   src/defineFunction.js:208:32
   208|         _functions[names[i]] = data;
                                       ^^^^

References:
   src/defineFunction.js:25:16
    25| ) => NodeValue<NODETYPE>;
                       ^^^^^^^^ [1]
   src/ParseNode.js:39:23
    39| export type NodeValue<TYPE: NodeType> = $ElementType<ParseNodeTypes, TYPE>;
                              ^^^^ [2]
```
Also, it seems `*`(existential type) is deprecated: https://github.com/facebook/flow/issues/6308, https://github.com/facebook/flow/issues/6294

Reply:
Actually, I'm happy with this deprecation, as I have a difficult time reasoning about what precise constraints the type system is enforcing -- at least in the way `*` is currently implemented. I've been looking at some way to represent `ParseNode<"accent">|ParseNode<"accentUnder">|...|ParseNode<"xarrow">` which is the actual thing we want whenever we write `ParseNode<*>`.

In the meantime, I'll take a look at the existing errors and see if I can resolve them. Thanks for pointing this out!

Reply:
These errors are resolved by https://github.com/Khan/KaTeX/pull/1369